File: Sample/java/src/main/java/net/nend/sample/java/banner/layoutpatterns/ListViewActivity.java
package net.nend.sample.java.banner.layoutpatterns;

import android.os.Bundle;
import android.widget.ArrayAdapter;
import android.widget.ListView;

import net.nend.android.NendAdView;
import net.nend.sample.java.SimpleListActivity;

import java.util.ArrayList;

/**
 * ListViewに表示するサンプル
 */
public class ListViewActivity extends SimpleListActivity {
    private NendAdView nendAdView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        ArrayList<String> list = new ArrayList<>();
        for (int i = 0; i < 100; i++) {
            list.add("item" + i);
        }

        ArrayAdapter<String> adapter = new ArrayAdapter<>(this,
                android.R.layout.simple_list_item_1, list);
        instantiateListAdapter(adapter);
    }

    @Override
    protected void onStart() {
        super.onStart();

        ListView listView = getListView();
        if (listView != null && nendAdView == null) {
            nendAdView = new NendAdView(this, 3174, "c5cb8bc474345961c6e7a9778c947957ed8e1e4f");
            nendAdView.loadAd();
            listView.addHeaderView(nendAdView);
        }
    }
}

File: Sample/kotlin/src/main/java/net/nend/sample/kotlin/banner/layoutpatterns/ListViewActivity.kt
package net.nend.sample.kotlin.banner.layoutpatterns

import android.os.Bundle
import android.widget.ArrayAdapter
import net.nend.android.NendAdView
import net.nend.sample.kotlin.SimpleListActivity
import net.nend.sample.kotlin.banner.BannerSampleActivity.Companion.BANNER_API_KEY_320_50
import net.nend.sample.kotlin.banner.BannerSampleActivity.Companion.BANNER_SPOT_ID_320_50

/**
 * ListViewに表示するサンプル
 */
class ListViewActivity : SimpleListActivity() {
    private var nendAdView: NendAdView? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val list = (0..99).map { i -> "item$i" }
        val listAdapter = ArrayAdapter(this, android.R.layout.simple_list_item_1, list)
        instantiateListAdapter(listAdapter)
    }

    override fun onStart() {
        super.onStart()

        nendAdView ?: run {
            nendAdView = NendAdView(this, BANNER_SPOT_ID_320_50, BANNER_API_KEY_320_50).apply {
                listFragment.listView.addHeaderView(this)
                loadAd()
            }
        }
    }
}